src/TwigComponent/src/HasAttributesTrait.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TwigComponent;
+
+use Symfony\UX\LiveComponent\Attribute\LiveProp;
+use Symfony\UX\TwigComponent\Attribute\PostMount;
+use Symfony\UX\TwigComponent\Attribute\PreMount;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @experimental
+ */
+trait HasAttributesTrait
+{
+    #[LiveProp(hydrateWith: 'hydrateAttributes', dehydrateWith: 'dehydrateAttributes')]
+    public ComponentAttributes $attributes;
+
+    #[PreMount]
+    public function preMountAttributes(array $data): array
+    {
+        $data['_attributes'] = $data['attributes'] ?? [];
```

Comment:
I'm missing the point of this. Is this to try to avoid `attributes` being available during the normal "mounting" process?

Reply:
Yes, this also gives the option to use the attributes sub-key to set attributes if you have a conflict with a property on your component (ie `$style`).

Reply:
I can't quite tell, are you thinking to remove the PreMount hook?

Reply:
I refactored this to remove the PreMount hook - much simpler now. Let me know if this is easier to understand.